File: app/src/main/java/io/bhurling/privatebet/Application.java
package io.bhurling.privatebet;

import android.content.Context;

public class Application extends android.app.Application {

    private ApplicationComponent mComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        if (mComponent == null) {
            mComponent = DaggerApplicationComponent.builder()
                    .applicationModule(new ApplicationModule(this))
                    .build();
        }
    }

    public static ApplicationComponent component(Context context) {
        return ((Application) context.getApplicationContext()).mComponent;
    }
}

File: app/src/main/java/io/bhurling/privatebet/Application.kt
package io.bhurling.privatebet

import dagger.hilt.android.HiltAndroidApp
import io.bhurling.privatebet.navigation.dryrun.DryRun
import timber.log.Timber
import timber.log.Timber.DebugTree

@HiltAndroidApp
class Application : android.app.Application() {

    override fun onCreate() {
        super.onCreate()

        if (BuildConfig.DEBUG) {
            Timber.plant(DebugTree())
        }

        if (BuildConfig.DEBUG) {
            DryRun.checkEntryPoints()
        }
    }
}
